x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/config_panel_wrapper.tsx

```diff
@@ -174,85 +190,329 @@ function LayerPanels(
+                          togglePopover: () => {
+                            if (popoverState.isOpen) {
+                              setPopoverState({
+                                isOpen: false,
+                                openId: null,
+                              });
+                            } else {
+                              setPopoverState({
+                                isOpen: true,
+                                openId: accessor,
+                              });
+                            }
+                          },
+                        }}
+                      />,
+                      <NativeRenderer
+                        render={props.datasourceMap[datasourceId].renderDimensionEditor}
+                        nativeProps={{
+                          ...layerDatasourceConfigProps,
+                          core: props.core,
+                          columnId: accessor,
+                          filterOperations: dimension.filterOperations,
+                        }}
+                      />
+                    )}
 
-      <EuiSpacer size="s" />
+                    <EuiButtonIcon
+                      data-test-subj="indexPattern-dimensionPopover-remove"
+                      iconType="cross"
+                      iconSize="s"
+                      size="s"
+                      color="danger"
+                      aria-label={i18n.translate('xpack.lens.indexPattern.removeColumnLabel', {
+                        defaultMessage: 'Remove configuration',
+                      })}
+                      title={i18n.translate('xpack.lens.indexPattern.removeColumnLabel', {
+                        defaultMessage: 'Remove configuration',
+                      })}
+                      onClick={() => {
+                        trackUiEvent('indexpattern_dimension_removed');
+                        props.updateVisualization(
+                          props.activeVisualization.removeDimension({
+                            layerId,
+                            dimensionId: dimension.dimensionId,
+                            columnId: accessor,
+                            prevState: props.visualizationState,
+                          })
+                        );
+                      }}
+                    />
+                  </DragDrop>
+                ))}
+                {dimension.supportsMoreColumns ? (
+                  <DragDrop
+                    className="lnsIndexPatternDimensionPanel"
+                    data-test-subj="indexPattern-dropTarget"
+                    droppable={
+                      dragDropContext.dragging &&
+                      layerDatasource.canHandleDrop({
+                        ...layerDatasourceDropProps,
+                        columnId: newId,
+                        filterOperations: dimension.filterOperations,
+                      })
+                    }
+                    onDrop={droppedItem => {
```

Comment:
For reviewers: Notice that the `onDrop` behavior has moved into the editor frame. Read this function carefully.